### PR TITLE
Handle case of multiple IP range, combination of CIDR and IP range

### PR DESCRIPTION
**Pull request by ShubhamRasal**
- Input may contain the multiple CIDRs and multiple IP ranges
  eg. `166.8.0.0/16,166.11.0.0/16,166.9.0.0-166.10.255.255`
- Handle the above input for arggregation, count, etc
- Move the string split logic from `main()` to `process()` function
- Add the new variable IPRangeMap which stores multiple IP range

### CODE REVIEW

File: cmd/mapcidr/main.go [hunk 1]
```diff
@@ -324,33 +333,27 @@ func process(wg *sync.WaitGroup, chancidr, outputchan chan string) {
...
 	if options.IPRangeInput {
-		// check if ipRange has more than 2 values
-		if len(ipRange) > 2 {
-			gologger.Fatal().Msgf("IP range can not have more than 2 values.")
-		}
-		cidrs, err := mapcidr.GetCIDRFromIPRange(ipRange[0], ipRange[1])
-		if err != nil {
-			gologger.Fatal().Msgf("%s\n", err)
-		}
+		for _, ipRange := range ipRangeMap {
```
Comment (jdk2588) [inline cmd/mapcidr/main.go]: You are not doing anything with the `key` of the map, so why need a map in first place?

File: cmd/mapcidr/main.go [hunk 2]
```diff
@@ -289,20 +282,36 @@ func prepareIPsFromCidrFlagList(items []string) []string {
...
-		ranger   *ipranger.IPRanger
-		err      error
-		hasSort  = options.SortAscending || options.SortDescending
-		ipRange  []net.IP
+		allCidrs     []*net.IPNet
+		pCidr        *net.IPNet
+		ranger       *ipranger.IPRanger
+		err          error
+		hasSort      = options.SortAscending || options.SortDescending
+		ipRangeMap   = make(map[int][]net.IP)
```
Comment (jdk2588) [inline cmd/mapcidr/main.go]: Why do we need a map, why can't the array work? Why do you need iprangeCount also?

**Review by jdk2588 — CHANGES_REQUESTED** [review]

File: cmd/mapcidr/main.go (hunk 1 above)
Comment (ShubhamRasal) [reply cmd/mapcidr/main.go]: Remove the map with the list

File: cmd/mapcidr/main.go (hunk 2 above)
Comment (ShubhamRasal) [reply cmd/mapcidr/main.go]: Removed the map with the list

**Review by jdk2588 — APPROVED** [review]

**Pull request merged**